### ISSUE TITLE
"The following build commands failed: Check dependencies"

### ISSUE THREAD

**Issue by yocontra** [remixz/ios-sticker-pack#1]
```
None of your accounts are a member of 'VA39VYN298':  An unexpected error occurred. Xcode cannot find a team matching 'VA39VYN298'.
No profiles for 'com.example.squad-stickers.squad-stickers.StickerPackExtension' were found:  Xcode couldn't find a provisioning profile matching 'com.example.squad-stickers.squad-stickers.StickerPackExtension'.
Code signing is required for product type 'App Extension' in SDK 'iOS 10.0'
```

Went into XCode and manually set it up to use my provisioning profile, still not working (same error).

**Comment by remixz** [comment]
😭  Thought I had scrubbed the template of my signing info. Now when I've actually removed them, the CLI wants to go into XCode to set it up. Hm... this might be a roadblock.

Can you try unchecking and rechecking the "Automatically manage signing" box, and then selecting your team? That should at least make it work for now.

**Comment by yocontra** [comment]
@remixz Same thing, same error.

**Comment by yocontra** [comment]
Key is still in this file, btw: https://github.com/remixz/ios-sticker-pack/blob/ff512829958013d6692f3730192f134705bf82bd/template/App/__APP_NAME__.xcodeproj/project.pbxproj (couple of places)

**Comment by remixz** [comment]
Dang... I'm not sure if this can be solved 😞  I'm trying to see if there's a way to read a certificate's info from the Keychain from node or a CLI, but I'm guessing it's unlikely, since that'd be a bit of a security issue. I'll keep looking. 

I'm surprised that disabling and reenabling the signing doesn't fix it either. Doing some diff's on the pbxproj files, it seems like Xcode sets some signing properties once and doesn't let you change them? ¯\_(ツ)_/¯

**Comment by remixz** [comment]
@contra Okay, if you don't mind helping out debugging... 😅  Can you try these steps when you get a chance?
1. Clone 2c6d58d and `npm link` it
2. Go to your Keychain Access app, find your iPhone Developer certificate (there's a search box at the top right), right click on it and choose "Get Info"
3. Scroll down in the window that popped up, and copy the "Organizational Unit" value
4. Run `ios-sticker-pack create --team="VALUE"`, replacing `VALUE` with whatever your "Organizational Unit" value was
5. Run `ios-sticker-pack deploy` and see if it works?

**Comment by remixz** [comment]
Oh, alternatively (and maybe progress towards a cleaner solution?), get the org unit value thingy, and then after running `ios-sticker-pack create` (no --team, so you could use the live version), run `xcodebuild -project App/squad-stickers.xcodeproj DEVELOPMENT_TEAM=VALUE`. Still have to figure out how to get that value pragmatically (I can get it from exporting the developer certificate, but I still have to do that manually from the Keychain Access app), but if I can do that, _maybe_ this is fixed?

**Comment by remixz** [comment]
Alright... my last update on this: I'm 99% sure this is unfixable with the current flow of 2 commands. :\ Even if I can get all the info from the cert, the Xcode app is the only way to contact Apple's servers to provision the app for your device. I realized this once I got the notification I'd provisioned too many apps. 10 a week... pff.

The one way I can think of fixing it is to require a step of opening Xcode, having the user manually create a blank stickers project, then using the CLI to write all the sticker metadata and deploy, but at that point... you could just drag and drop the images in Xcode, and deploy from there, no code required. I guess it might be a little faster since you'd only have to open Xcode once no matter how many times you want to update your stickers, but... meh, I feel like constant updates are a bit of an edge case.

Overall, I'm just kinda disappointed!  This has served as a good reminder of why I stuck with web development all this time. 😂  I'll leave this repo up of course, but it's probably only useful for me at this point. At least it let me create this, which I'm sure I can put in a Slack team at some point:

![sticka](https://cloud.githubusercontent.com/assets/1459603/18698499/884502b2-7f7f-11e6-9029-bf1e4df3df53.png)

**Comment by yocontra** [comment]
@remixz How about this:
- Remove all signing keys, dev teams, etc. from the template so they are completely unset
- Running `create` also opens the project up and logs instructions on how to set your developer team

Really the issue isn't that you have to open xcode and click a switch or two, it's that the keys are hardcoded in the template so you can't unset them. Removing them from the template + having simple instructions on how to set the signing key should be all that is needed, right?

**Comment by yocontra** [comment]
BTW - even if you wanted to automate the whole thing, you totally could. I've had a setup using https://fastlane.tools/ + cordova where I never had to open xcode to mess with signing keys, builds, app store publishes, etc.

Some tools that could help out with this:
- https://github.com/fastlane/fastlane/tree/master/sigh (fixing certs)
- https://github.com/fastlane/fastlane/tree/master/cert (finding local certs + creating new ones)
- https://github.com/fastlane/fastlane/tree/master/gym (building with better cert support)

**Comment by remixz** [comment]
@contra I tried doing that last night, but the problem was that when I removed all the signing info from the template and tried to set it in Xcode, Xcode would only set up signing for the app bundle, and not the extension. From what I could tell (and I guess this is an Xcode bug?), the only way to set the signing info for the extension was through creating a new sticker pack project in Xcode.

I guess I could do that flow though, except it'd be something like "gonna open Xcode, here's how to set your team, press enter when you've done this", then it'd look at the signing info set for the app bundle in the pbxproj file, and copy it to the sections for the extension. I'll give it a try.

**Comment by yocontra** [comment]
@remixz Did none of the fastlane tools seem like they could help with this?

**Comment by remixz** [comment]
@contra I just tried it now, but `sigh` wouldn't let me use my free developer account. Looks like it's a known issue too: https://github.com/fastlane/fastlane/issues/6022

**Comment by remixz** [comment]
@contra Okay, I think I've got it working!! 🎉  🎉  🎉  Can you clone https://github.com/remixz/ios-sticker-pack/commit/bc14fe6ed736b45879d89dc168dd301361f6ba4d and try it out with the instructions? Make sure to delete the old `App` folder first. It works for me, just want to make sure it works on one other machine before I release it.

**Comment by yocontra** [comment]
@remixz Works for me, getting a diff issue now (`No device was connected within 5 seconds -- aborting`, probably unrelated)

**Comment by remixz** [comment]
@contra Ah, I was worried about that... the app built fast for me, so I set a 5 second timeout in case I forgot to connect my phone, but I think it might just cause issues if it doesn't build that fast for some reason. I've removed that, and just put a reminder to connect the device.

Thanks so much for your help!! Really appreciate it. 😄  I've released the fixed version as 1.1.0.

**Comment by lydia413** [comment]
(does commenting reopen this issue?)
i'm seeing this same problem with the current version, 1.1.1: i can create successfully, but not deploy. xcode appears to believe that all is well with my provisioning profile. should all be well for iOS 11?
